### PR TITLE
feat(testing): allow to disable network error logging via 'logFailingNetworkRequests' option

### LINKED ISSUE #2572: Option to disable logging failed requests during E2E tests
**Stencil version:** 1.16.0

**I'm submitting a:**

[ ] bug report
[X] feature request
[ ] support request 

**Current behavior:**

Currently it defaults to logging errors to the console when requests abort/fail in E2E tests.

**Expected behavior:**

I'm writing tests for a video player and sometimes there are requests lingering at the end of my tests from buffering frames and other stuff. I'd like an option when creating `newE2EPage` to to stop the errors from logging.

Maybe an option to disable requests that were simply aborted `req.failure()?.errorText === 'net::ERR_ABORTED'` would be more specific?

**Related code:**

https://github.com/ionic-team/stencil/blob/b47093b2f86be777c39f79abc691a8a6d1ec50d7/src/testing/puppeteer/puppeteer-page.ts#L141

### PR DESCRIPTION
**Pull request by christian-bromann**
## What is the current behavior?

fixes #2572

## What is the new behavior?
This patch allows users to change the behavior to not log failing network requests.

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Testing

n/a

## Other information

n/a

### CODE REVIEW

**Comment by christian-bromann** [comment]
Raised a PR with extended docs in https://github.com/ionic-team/stencil-site/pull/1448

File: src/testing/puppeteer/puppeteer-page.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ export async function newE2EPage(opts: NewE2EPageOptions = {}): Promise<E2EPage>
 
     const failOnConsoleError = opts.failOnConsoleError === true;
     const failOnNetworkError = opts.failOnNetworkError === true;
+    const logFailingNetworkRequests = opts.logFailingNetworkRequests === true;
```
Comment (tanner-reits) [inline src/testing/puppeteer/puppeteer-page.ts:118]: Should we keep this as the default behavior (i.e. make it opt out)?

**Review by tanner-reits — APPROVED** [review]
One thought

File: src/testing/puppeteer/puppeteer-page.ts (hunk 1 above)
Comment (christian-bromann) [reply src/testing/puppeteer/puppeteer-page.ts:118]: ah yes, that was my intention: keep the current behavior

**Pull request merged**